components/ws-daemon/nsinsider/main.go

```diff
@@ -528,3 +562,17 @@ const (
 	// FlagAtRecursive: Apply to the entire subtree: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/fcntl.h#L112
 	flagAtRecursive = 0x8000
 )
+
+func addConnectionLimitRule(limit int) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+	rule := fmt.Sprintf("add rule ip gitpod ratelimit ip protocol tcp ct state new meter connmeter { ip daddr & 0.0.0.0 timeout 1m limit rate over %d/minute } counter drop", limit)
+
+	// the go library does not support meters, so use nft instead
+	cmd := exec.CommandContext(ctx, "/usr/sbin/nft", rule)
```

Comment:
Can you explain what kind of issues have been observed in the context of kubelet? I was not able to find anything.

Reply:
@aledbf Are you ok with merging this or would you like to discuss it further?

Reply:
I am not sure we should merge this without knowing the behavior of different `nftable` versions and if there is any conflict with the CNI bandwidth plugin.

Did you test this change in an ephemeral cluster?

Reply:
workspace preview

Reply:
We used nftables to replace slirp4net, if there is an issue with versions, would we have more fundamental issues with workspace networking? 
* This makes me wonder if we should add related [runtime tests for networking](https://www.notion.so/gitpod/Common-runtime-tests-4e3b05c3b81e4ff6810572aee66ffad3). 
* I ask because I'm not sure what dimensions to consider.

@aledbf I like the idea of testing with an ephemeral cluster. Can you [add a related test here](https://www.notion.so/gitpod/Common-runtime-tests-4e3b05c3b81e4ff6810572aee66ffad3), so we can have a repeatable way to do in the future? 

@Furisto let's leave on-hold till we've had a chance to test in an ephemeral cluster (to test for impacts to CNI bandwidth plugin).